### DISCUSSION TITLE
Not found card when run command " opensc-tool -a"

### DISCUSSION THREAD

**Discussion by GitHubDuong** [polhenarejos/pico-hsm#49] [General]
I completed the build when run the command " opensc-tool -a" and got a message "No smart card readers found". currently I use pico RP2040.

**Comment by polhenarejos** [comment]
Did you patch VID/PID?

**Reply by GitHubDuong** [reply]
yep. use this tool to path: https://www.picokeys.com/pico-patcher/
![image](https://github.com/user-attachments/assets/97d795fe-64f7-457e-8e72-5c8c80535c7e)

I' using hardware 
https://www.raspberrypi.com/products/raspberry-pi-pico/

Build Version : [pico_hsm_pico-4.0-eddsa1.uf2](https://github.com/polhenarejos/pico-hsm/releases/download/v4.0-eddsa1/pico_hsm_pico-4.0-eddsa1.uf2)

**Reply by polhenarejos** [reply]
Which VID is this one? You have to use one of those supported by libccid. Here's a list:
https://ccid.apdu.fr/select_readers/

**Reply by GitHubDuong** [reply]
Thanks  polhenarejos. 
I can connect pico with opensc but I have another issue with pcks11-tools: 

![image](https://github.com/user-attachments/assets/7413cf33-74c6-493b-8975-db7c9cbba91c)

**Reply by polhenarejos** [reply]
The first thing is initialize the HSM via `python pico-hsm-tool.py initialize`.

**Reply by GitHubDuong** [reply]
pkcs11-tool not supported encrypt. right?
![image](https://github.com/user-attachments/assets/3a6a3146-842d-48e2-a76e-5740bd3ec0c4)

**Reply by polhenarejos** [reply]
I use this command:
`sc_tool --encrypt --id 1 --input-file test --mechanism aes-cbc > crypted.aes 2>/dev/null`

**Reply by GitHubDuong** [reply]
it's not the same as your document.  
Can you provide me with the new document?
![image](https://github.com/user-attachments/assets/c668e2aa-9705-4099-b8e2-cad7928644c1)
